src/Compilers/CSharp/Test/Syntax/Parsing/RawInterpolatedStringLiteralCompilingTests.cs

```diff
@@ -0,0 +1,1774 @@
+                //  a
+                Diagnostic(ErrorCode.ERR_LineDoesNotStartWithSameWhitespace, " ").WithLocation(5, 1));
+    }
+
+    [Fact]
+    public void MultiLineCase54()
+    {
+        RenderAndVerify(@"
+System.Console.Write(
+    $""""""
+    a
+    a
+    {42}
+    b
+    b
+    a{43}
+    c
+␠
+    c
+    """""");", expectedOutput: @"a
+a
+42
+b
+b
+a43
+c
+
+c");
+    }
+
+    [Fact]
+    public void MultiLineCase55()
+    {
+        RenderAndVerify(@"
+System.Console.Write(
+    $""""""
+    a
+    a
+    {42}
+    b
+    b
+    a{43}
+    c
+␠␠␠␠␠
+    c
+    """""");", expectedOutput: @"a
+a
+42
+b
+b
+a43
+c
+␠
+c");
+    }
+
+    [Fact]
+    public void MultiLineCase56()
+    {
+        RenderAndVerify(@"
+System.Console.Write(
+    $""""""
+    a
+    a
+    {42}
+    b
+    b
+    a{43}
+    c
+␠c
+    c
+    """""");",
+                // (11,1): error CS9103: Line does not start with the same whitespace as the closing line of the raw string literal
+                //  c
+                Diagnostic(ErrorCode.ERR_LineDoesNotStartWithSameWhitespace, " ").WithLocation(11, 1));
+    }
+
+    [Fact]
+    public void MultiLineCase57()
+    {
+        RenderAndVerify(@"
+System.Console.Write(
+    $""""""
+
+    """""");", expectedOutput: "");
+    }
+
+    [Fact]
+    public void MultiLineCase58()
+    {
+        RenderAndVerify(@"
+System.Console.Write(
+    $""""""
+␠
+    """""");", expectedOutput: "");
+    }
+
+    [Fact]
+    public void MultiLineCase59()
+    {
+        RenderAndVerify(@"
+System.Console.Write(
+    $""""""
+␠␠
+    """""");", expectedOutput: "");
+    }
+
+    [Fact]
+    public void MultiLineCase60()
+    {
+        RenderAndVerify(@"
+System.Console.Write(
+    $""""""
+␠␠␠␠
+    """""");", expectedOutput: "");
+    }
+
+    [Fact]
+    public void MultiLineCase61()
+    {
+        RenderAndVerify(@"
+System.Console.Write(
+    $""""""
+␠␠␠␠␠
+    """""");", expectedOutput: "␠");
+    }
+
+    [Fact]
+    public void MultiLineCase62()
+    {
+        RenderAndVerify(@"
+System.Console.Write(
+    $""""""
+    a
+
+    """""");", expectedOutput: @"a
+");
+    }
+
+    [Fact]
+    public void MultiLineCase63()
+    {
+        RenderAndVerify(@"
+System.Console.Write(
+    $""""""
+    a
+␠
+    """""");", expectedOutput: @"a
+");
+    }
+
+    [Fact]
+    public void MultiLineCase64()
+    {
+        RenderAndVerify(@"
+System.Console.Write(
+    $""""""
+    a
+␠␠
+    """""");", expectedOutput: @"a
+");
+    }
+
+    [Fact]
+    public void MultiLineCase65()
+    {
+        RenderAndVerify(@"
+System.Console.Write(
+    $""""""
+    a
+␠␠␠␠
+    """""");", expectedOutput: @"a
+");
+    }
+
+    [Fact]
+    public void MultiLineCase66()
+    {
+        RenderAndVerify(@"
+System.Console.Write(
+    $""""""
+    a
+␠␠␠␠␠
+    """""");", expectedOutput: @"a
+␠");
+    }
+
+    [Fact]
+    public void TestOutVarOrderOfEvaluation()
```

Comment:
Consider adding one for the reverse order as well (should error)

Reply:
added (with error).